testsuite/tests/typing-warnings/unused_types.ml

```diff
@@ -671,3 +671,21 @@ Warning 34 [unused-type-declaration]: unused type "unused2".
 
 val f : 'a -> 'a = <fun>
 |}]
+
+let f (x1 : int as 'unused1) (x2 : int as 'unused2)
+      (x3 : int as 'used1) (x4 : 'used1) (x5 : 'used2) = (x1, x2, x3, x4, x5)
+
+[%%expect{|
+Line 1, characters 12-27:
+1 | let f (x1 : int as 'unused1) (x2 : int as 'unused2)
+                ^^^^^^^^^^^^^^^
+Warning 34 [unused-type-declaration]: unused type "'unused1".
+
+Line 1, characters 35-50:
+1 | let f (x1 : int as 'unused1) (x2 : int as 'unused2)
+                                       ^^^^^^^^^^^^^^^
+Warning 34 [unused-type-declaration]: unused type "'unused2".
```

Comment:
The error message is starting to stretch the meaning of `type`. One option would be to add a `type_declaration_kind` variant to the `Unused_type_declaration` constructor in order to specialize the error messages. Another would to split the new warnings.